### PR TITLE
Fix Replayability 

### PR DESCRIPTION
**Pull request by rakrakon**
- **Change Intake class to private constructor**
- **Create IO's map in Initializer**
- **Create initializeSubsystems function in initializer**
- **Change Robot to use initializeSubsystems function**
- **Remove current mode parameter from initSubsystems function**
- **Add initVision & initSwerve in initSubsytems function**
- **Optimize imports**

### CODE REVIEW

**Comment by katzuv** [comment]
Change the PR title to be in English and not in Gittish.

File: src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt [hunk 1]
```diff
@@ -1,69 +1,66 @@
...
+
+
+    init {
+        VisionSim.system.addAprilTags(VisionConstants.aprilTagFieldLayout)
+        VisionSim.system.addCamera(simCamera, robotToCam)
+    }
+
+    private fun pose2dToPose3d(pose: Pose2d): Pose3d = Pose3d(
+        pose.x, pose.y, 0.0, Rotation3d(0.0, 0.0, pose.rotation.radians)
+    )
```
Comment (katzuv) [inline src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt]: 1. Doesn't belong here
2. Should be an extension function

File: src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt [hunk 2]
```diff
@@ -1,69 +1,66 @@
...
+                0.0,
+                botPose3d + robotToCam.inverse(),
+                VisionConstants.aprilTagFieldLayout.tags.map { a ->
+                    VisionTargetSim(
+                        a.pose,
+                        TargetModel.kAprilTag36h11,
+                        a.ID
+                    )
+                }
+            )
```
Comment (katzuv) [inline src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt]: This is too long

File: src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt [hunk 3]
```diff
@@ -1,69 +1,66 @@
...
+    )
+
+    override fun updateInputs() {
+        val botPose = SwerveDrive.getInstance().estimator.estimatedPosition
+        val botPose3d = pose2dToPose3d(botPose)
+        val latestResult =
+            simCamera.process(
+                0.0,
+                botPose3d + robotToCam.inverse(),
+                VisionConstants.aprilTagFieldLayout.tags.map { a ->
```
Comment (katzuv) [inline src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt]: 1. If you use `a`, you can simply use the default `it`.
2. I believe `forEach` fits here better.

**Review by katzuv — CHANGES_REQUESTED** [review]

File: src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt (hunk 1 above)
Comment (rakrakon) [reply src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt]: a0d3a7d44325eb79f90c2aec2505aeca24756d43

File: src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt (hunk 3 above)
Comment (rakrakon) [reply src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt]: 419a0a1154b92f968f05e16059be28c5442998eb

File: src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt (hunk 2 above)
Comment (rakrakon) [reply src/main/kotlin/frc/robot/subsystems/vision/PhotonVisionIOSim.kt]: 2acf7c1bfeeb1873a469735016a8f158691ab19e

**Review by katzuv — APPROVED** [review]

**Pull request merged**